xarray/backends/h5netcdf_.py

```diff
@@ -49,49 +82,60 @@ class H5NetCDFStore(WritableCFDataStore, DataStorePickleMixin):
     """Store for reading and writing data via h5netcdf
     """
     def __init__(self, filename, mode='r', format=None, group=None,
-                 writer=None):
+                 writer=None, autoclose=False):
         if format not in [None, 'NETCDF4']:
             raise ValueError('invalid format for h5netcdf backend')
         opener = functools.partial(_open_h5netcdf_group, filename, mode=mode,
                                    group=group)
         self.ds = opener()
+        if autoclose:
+            warnings.warn('autoclose=True is disabled for the h5netcdf backend '
```

Comment:
@shoyer, do you want to explicitly let this go on without an error?  It may catastrophically error out if we just attempt to do `autoclose=True`.  My preference would be to use a `NotImplemented` error.